src/com/palmergames/bukkit/towny/war/siegewar/utils/SiegeWarBlockUtil.java

```diff
@@ -0,0 +1,151 @@
+package com.palmergames.bukkit.towny.war.siegewar.utils;
+
+import com.palmergames.bukkit.towny.TownySettings;
+import com.palmergames.bukkit.towny.TownyUniverse;
+import com.palmergames.bukkit.towny.exceptions.NotRegisteredException;
+import com.palmergames.bukkit.towny.object.Coord;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import com.palmergames.bukkit.towny.object.TownyWorld;
+import com.palmergames.bukkit.towny.war.siegewar.enums.SiegeStatus;
+import com.palmergames.bukkit.towny.war.siegewar.locations.SiegeZone;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.Player;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Goosius
+ */
+public class SiegeWarBlockUtil {
+	public static List<TownBlock> getAdjacentTownBlocks(Player player, Block block) {
+		TownyUniverse townyUniverse = TownyUniverse.getInstance();
+		TownyWorld townyWorld;
+		List<TownBlock> nearbyTownBlocks = new ArrayList<>();
+
+		try {
+			townyWorld = townyUniverse.getDataSource().getWorld(player.getWorld().getName());
+		} catch (NotRegisteredException e) {
+			return nearbyTownBlocks;
+		}
+
+		List<Coord> nearbyCoOrdinates =new ArrayList<>();
+		Coord blockCoordinate = Coord.parseCoord(block);
+		nearbyCoOrdinates.add(blockCoordinate.add(0,-1));
+		nearbyCoOrdinates.add(blockCoordinate.add(0,1));
+		nearbyCoOrdinates.add(blockCoordinate.add(1,0));
+		nearbyCoOrdinates.add(blockCoordinate.add(-1,0));
+
+		TownBlock nearbyTownBlock = null;
+		for(Coord nearbyCoord: nearbyCoOrdinates){
+			if(townyWorld.hasTownBlock(nearbyCoord)) {
+
+				try {nearbyTownBlock = townyWorld.getTownBlock(nearbyCoord);
+				} catch (NotRegisteredException e) {}
+
+				if (nearbyTownBlock.hasTown()) {
+					nearbyTownBlocks.add(nearbyTownBlock);
+				}
+			}
+		}
+
+		return nearbyTownBlocks;
+	}
+
+	public static boolean doesPlayerHaveANonAirBlockAboveThem(Player player) {
+		return doesLocationHaveANonAirBlockAboveIt(player.getLocation());
+	}
+
+	public static boolean doesBlockHaveANonAirBlockAboveIt(Block block) {
+		return doesLocationHaveANonAirBlockAboveIt(block.getLocation());
+	}
+
+	private static boolean doesLocationHaveANonAirBlockAboveIt(Location location) {
+		location.add(0,1,0);
+
+		while(location.getY() < 256)
+		{
+			if(location.getBlock().getType() != Material.AIR)
+			{
+				return true;   //There is a non-air block above them
+			}
+			location.add(0,1,0);
+		}
+		return false;  //There is nothing but air above them
+	}
+
+	public static boolean isBlockNearAnActiveSiegeBanner(Block block) {
+		//Look through all siege zones
```

Comment:
These single-line comments should really be combined into a multi-line, or converted into a Javadoc comment.

Reply:
Compressed comment into javadoc..